### PR TITLE
Add contact-list-item styles

### PR DESCRIPTION
**Pull request by Ponomaleks**

### CODE REVIEW

File: src/app/shared/pipes/age.ts
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/shared/pipes/age.ts:1]: added pipe.ts suffix

File: src/app/main/components/contact-list-item/contact-list-item.component.html [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+      </a>
+    </li>
+  </ul>
+</div>
+<epm-button
+  class="message-btn"
+  [class]="[BUTTON_THEMES.Primary, BUTTON_THEMES.Small, BUTTON_THEMES.IconOnly]"
+  [iconName]="ICON_NAMES.Chat"
+  (click)="onChatClick($event)"
+></epm-button>
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.html]: fixed

File: src/app/shared/pipes/age.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: 'age',
+  standalone: true
+})
+export class Age implements PipeTransform {
+  transform(birthdate: string): number {
+    const SEPARATOR = '.';
+    const birthdateParts: string[] = birthdate.split(SEPARATOR);
+    const birthYear: number = +birthdateParts[2];
+    const birthMonth: number = +birthdateParts[1];
+    const birthDay: number = +birthdateParts[0];
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/shared/pipes/age.ts]: destructurisation was made

File: src/app/main/pages/contact-list/contact-list.component.ts [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
   standalone: true,
-  imports: [CommonModule, ChatListItemComponent, ContactListItemComponent],
+  imports: [CommonModule, ContactListItemComponent],
   templateUrl: './contact-list.component.html',
   styleUrls: ['./contact-list.component.scss']
 })
-export class ContactListComponent {
-  public users: UserDetailed[] = USERS;
+export class ContactListComponent implements OnInit, OnDestroy {
+  userList: Signal<ExpandedUserDetailed[]> = this.mainApiService.userList;
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.ts]: value assignment was moved to OnInit hook

File: src/app/main/main.config.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export const SOCIAL_PROFILE_LINKS_ROOT: { [key: string]: (userName: string) => string } = {
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/main.config.ts]: done

File: src/app/main/components/contact-list-item/contact-list-item.component.ts [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+  onChatClick(event: MouseEvent): void {
+    event.stopPropagation();
+    this.router.navigateByUrl(`${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Chat}/${this.user.userId}`, {
+      state: this.user
+    });
+  }
+
+  onSocialClick(event: MouseEvent, link: string, type: SOCIAL_ICONS): void {
+    const target = type === SOCIAL_ICONS.Skype ? '_self' : '_blank';
+    event.stopPropagation();
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.ts:35]: we decide not to separate stopPropagation yet

File: src/app/main/components/contact-list-item/contact-list-item.component.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
+@import 'variables';
+@import 'mixins';
+
+:host {
+  position: relative;
+  display: flex;
+  padding: 16px 16px 20px 16px;
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.scss]: fixed

File: src/app/main/components/contact-list-item/contact-list-item.component.html [hunk 7]
```diff
@@ -1,5 +1,26 @@
...
+<div class="avatar">
+  <div class="avatar-wrapper">
+    <img width="48" height="48" [ngSrc]="user.avatarUrl" alt="user avatar" />
+  </div>
 </div>
+<div class="info">
+  <div class="info-text">
+    <b class="name"
+      >{{ user.name }} {{ user.surname }}<span *ngIf="user.dateOfBirth">, {{ user.dateOfBirth | age }}</span>
+    </b>
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/components/contact-list-item/contact-list-item.component.html]: dine

File: src/app/shared/pipes/age.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    if (
+      currentDate.getMonth() < +birthMonth ||
+      (currentDate.getMonth() === +birthMonth && currentDate.getDate() < +birthDay)
+    ) {
+      return age - 1;
+    }
+
+    return age;
+  }
+}
```
Comment (GennadiyShpak) [inline src/app/shared/pipes/age.pipe.ts:24]:
```suggestion
export class AgePipe implements PipeTransform {
  transform(birthdate: string): number {
    const SEPARATOR = '.';
    const [birthDay, birthMonth, birthYear] = birthdate.split(SEPARATOR).map(item => Number(item));

    const currentDate: Date = new Date();
    const age: number = currentDate.getFullYear() - birthYear;
    const currentMonth = currentDate.getMonth()

    if (
      currentMonth  <  birthMonth ||
      (currentMonth  ===  birthMonth && currentDate.getDate() < birthDay)
    ) {
      return age - 1;
    }

    return age;
  }
}
```

File: src/app/main/pages/contact-list/contact-list.component.scss [hunk 9]
```diff
@@ -0,0 +1,7 @@
+.contact-list {
+  padding: 25px 0;
+
+  &-item {
+    margin-bottom: 12px;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/main/pages/contact-list/contact-list.component.scss]: changed for BEM calming

File: src/app/shared/pipes/age.pipe.ts (hunk 8 above)
Comment (Ponomaleks) [reply src/app/shared/pipes/age.pipe.ts]: Thanks, changed as suggested

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**